zerver/views/user_groups.py

```diff
@@ -115,6 +119,54 @@ def update_user_group_backend(
     return json_success(request, data)
 
 
+def send_messages_for_user_groups(
+    acting_user: UserProfile,
+    recipient_users: List[UserProfile],
+    user_group_name: str,
+    *,
+    send_subscription_message: bool,
+    send_unsubscription_message: bool,
+) -> None:
+    realm = acting_user.realm
+    mention_backend = MentionBackend(realm.id)
+
+    notifications = []
+    sender = get_system_bot(settings.NOTIFICATION_BOT, realm.id)
+    for recipient_user in recipient_users:
+        if recipient_user.email == acting_user.email:
```

Comment:
Never compare emails -- you should compare the objects' IDs, since those are guaranteed to be unique.